### PR TITLE
Extract the FemtoVG renderer into a separate crate

### PR DESCRIPTION
**Pull request by tronical**
This will be needed for a future experiment. Unlike the Skia renderer, which operates on raw window handles, the FemtoVG renderer exposes a different interface where it assumes that the caller takes care of the OpenGL context state. This means more boilerplate remains in the winit backend, including the glutin dependency. The upside is that it will allow using the FemtoVG renderer in environments without glutin.

### CODE REVIEW

**Comment by tronical** [comment]
This is deliberately still a draft and work in progress. One of the things missing is internal documentation.

**Comment by ogoffart** [comment]
(don't forget to add the entry in the publish.sh)

File: internal/renderers/femtovg/lib.rs [hunk 1]
```diff
@@ -0,0 +1,417 @@
+// Copyright © SixtyFPS GmbH <info@slint-ui.com>
+// SPDX-License-Identifier: GPL-3.0-only OR LicenseRef-Slint-commercial
+
```
Comment (ogoffart) [inline internal/renderers/femtovg/lib.rs:3]: There should be a crate documentation like for the other internal backend.

File: internal/renderers/femtovg/build.rs [hunk 2]
```diff
@@ -0,0 +1,19 @@
+// Copyright © SixtyFPS GmbH <info@slint-ui.com>
+// SPDX-License-Identifier: GPL-3.0-only OR LicenseRef-Slint-commercial
+
+use cfg_aliases::cfg_aliases;
+
+fn main() {
+    // Setup cfg aliases
+    cfg_aliases! {
```
Comment (ogoffart) [inline internal/renderers/femtovg/build.rs:8]: Not an objection to the patch, but while i think cfg_alias is great, having a build.rs just for that might have a significant impact on compile time.  I haven't measured it though. So I don't know if it is worth it or not.

File: internal/backends/winit/Cargo.toml [hunk 3]
```diff
@@ -21,7 +21,7 @@ path = "lib.rs"
 [features]
 wayland = ["winit/wayland", "glutin/wayland", "copypasta/wayland", "i-slint-renderer-skia?/wayland"]
 x11 = ["winit/x11", "glutin/x11", "glutin/glx", "copypasta/x11", "i-slint-renderer-skia?/x11"]
-renderer-winit-femtovg = ["femtovg", "fontdb", "libc", "yeslogic-fontconfig-sys", "winapi", "dwrote", "imgref", "unicode-script", "ttf-parser", "rgb", "i-slint-core/box-shadow-cache"]
+renderer-winit-femtovg = ["i-slint-renderer-femtovg"]
```
Comment (ogoffart) [inline internal/backends/winit/Cargo.toml:24]:
```suggestion
renderer-winit-femtovg = ["dep:i-slint-renderer-femtovg"]
```

But if we do that, we should make it consistent with other features

**Review by ogoffart — APPROVED** [review]

File: internal/renderers/femtovg/build.rs (hunk 2 above)
Comment (tronical) [reply internal/renderers/femtovg/build.rs:8]: There are 7 uses of `#[cfg(use_fontconfig)]`. I could replace those with the full expression, although I think that makes the code less readable. I'll post a patch on top and then we can see.

File: internal/renderers/femtovg/lib.rs (hunk 1 above)
Comment (tronical) [reply internal/renderers/femtovg/lib.rs:3]: Ah yes, I added docs for the main type, but not the library create. Will add.

File: internal/backends/winit/Cargo.toml (hunk 3 above)
Comment (tronical) [reply internal/backends/winit/Cargo.toml:24]: I don't feel very strong about this, but it sounds like something that should be applied to all crates. That said, I don't mind if we don't use dep: everywhere - at least so far I haven't been confused by the mix :)

**Pull request merged**